### PR TITLE
Use consistent NaN behavior for max functions

### PR DESCRIPTION
**Pull request by electrum**
NaN is effectively ignored, unless it is the only value. In other words, it is treated as the maximum value for `min`, and the minimum value for `max`.

From an implementation point of view, since it normally compares as the maximal value, we only need to special case it for maximum by reversing its comparator.

### CODE REVIEW

File: presto-main/src/test/java/io/prestosql/operator/scalar/TestMathFunctions.java [hunk 1]
```diff
@@ -1169,15 +1169,23 @@ public void testGreatest()
...
-        // float
-        assertFunction("greatest(REAL '1.5', 2.3E0)", DOUBLE, 2.3);
-        assertFunction("greatest(REAL '-1.5', -2.3E0)", DOUBLE, (double) -1.5f);
-        assertFunction("greatest(-1.5E0, REAL '-2.3', -5/3)", DOUBLE, -1.0);
-        assertFunction("greatest(REAL '1.5', REAL '-1.0' / 0.0E0, 1.0E0 / REAL '0.0')", DOUBLE, (double) (1.0f / 0.0f));
-        assertFunction("greatest(5, REAL '4', CAST(NULL as DOUBLE), 3)", DOUBLE, null);
+        assertFunction("greatest(NaN(), 5, 4, 3)", DOUBLE, 5.0);
+        assertFunction("greatest(5, 4, NaN(), 3)", DOUBLE, 5.0);
+        assertFunction("greatest(5, 4, 3, NaN())", DOUBLE, 5.0);
+        assertFunction("greatest(NaN())", DOUBLE, Double.NaN);
```
Comment (dain) [inline presto-main/src/test/java/io/prestosql/operator/scalar/TestMathFunctions.java:1177]: maybe add a test with multiple NaNs

**Review by dain — APPROVED** [review]
Looks good.

**Comment by findepi** [comment]
Title talks about max, i see you;re changing greatest. Are you taking care of `array_{min,max}` and `{min,max}_by` as well?
after the change, Is `min` going to be produce same result as `array_min(array_agg(...))` , and same result as `reduce(array_agg(...),  ... -> least(...)`?

**Comment by electrum** [comment]
I consider `greatest` to be the scalar version of `max`. It should touch all of the `max` related functions, unless I missed any. Those should all produce the same result -- that's the goal of making these consistent.

Note that the NULL behavior is different, since `array_min` / `array_max` returns NULL if any element is null. The same is true for `least` / `greatest`.

**Pull request merged**